container/libcontainer/helpers.go

```diff
@@ -163,17 +159,42 @@ func scanInterfaceStats(netStatsFile string) ([]info.InterfaceStats, error) {
 
 	scanner := bufio.NewScanner(file)
 
+	// Discard header lines
+	for i := 0; i < 2; i++ {
+		b := scanner.Scan()
+		if !b {
+			return stats, scanner.Err()
+		}
+	}
+
 	for scanner.Scan() {
 		line := scanner.Text()
 		line = strings.Replace(line, ":", "", -1)
 
-		i := info.InterfaceStats{}
+		fields := strings.Fields(line)
+		// If the format of the  line is invalid then don't trust any of the stats
+		// in this file.
+		if len(fields) != 17 {
+			return []info.InterfaceStats{}, fmt.Errorf("invalid interface stats line: %v", line)
+		}
+
+		devName := fields[0]
+		if !isIgnoredDevice(devName) {
+			i := info.InterfaceStats{
+				Name: devName,
+			}
 
-		_, err := fmt.Sscanf(line, netstatsLine,
-			&i.Name, &i.RxBytes, &i.RxPackets, &i.RxErrors, &i.RxDropped, &bkt, &bkt, &bkt,
-			&bkt, &i.TxBytes, &i.TxPackets, &i.TxErrors, &i.TxDropped, &bkt, &bkt, &bkt, &bkt)
+			statFields := append(fields[1:5], fields[9:13]...)
+			statPointers := []*uint64{
+				&i.RxBytes, &i.RxPackets, &i.RxErrors, &i.RxDropped,
+				&i.TxBytes, &i.TxPackets, &i.TxErrors, &i.TxDropped,
+			}
+
+			err := setInterfaceStatValues(statFields, statPointers)
+			if err != nil {
+				return []info.InterfaceStats{}, fmt.Errorf("cannot parse interface stats (%v): %v", err, line)
```

Comment:
Just curious, but why not return a nil slice? We didn't before, but it should be fine.